### PR TITLE
Add an option to report success in jira

### PR DESCRIPTION
**Pull request by rnetser**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ The firewatch project is in its infancy and currently has limited functionality.
   - If no failures are found, firewatch will search for any open issues on the Jira server provided and add a comment to the issue mentioning that the job has passed since that issue was filed.
     - This functionality also uses the labels on issues created by firewatch.
     - **Note:** If you get a notification on an issue, but would like to continue working on the issue without getting notifications, add the `ignore-passing-notification` label to the issue.
+  - If firewatch config contains job success reporting, a jira will be created reporting the job success
```
Comment (myakove) [inline README.md:23]: You should mention that the ticket status will be closed.

File: cli/objects/jira_base.py [hunk 2]
```diff
@@ -141,6 +143,13 @@ def create_issue(
             self.assign_issue(user_email=assignee, issue=issue.key)
             self.logger.info(f"Issue {issue} has been assigned to user {assignee}")
 
+        if close_issue:
+            self.connection.transition_issue(
+                issue=issue.key,
+                transition="closed",
+                comment="closed by firewatch",
```
Comment (myakove) [inline cli/objects/jira_base.py:150]: Maybe link to the repository in the comment as well?

File: cli/report/__init__.py [hunk 3]
```diff
@@ -94,5 +94,9 @@ def report(
...
     )
 
-    # Build the Report object and report issues to Jira
-    Report(firewatch_config=config, job=job)
+    try:
+        # Build the Report object and report issues to Jira
+        Report(firewatch_config=config, job=job)
+
+    finally:
+        job.delete_job_dir()
```
Comment (myakove) [inline cli/report/__init__.py:102]: What is job_dir?

File: cli/report/report.py [hunk 4]
```diff
@@ -60,6 +60,19 @@ def __init__(self, firewatch_config: Configuration, job: Job) -> None:
                 exit(1)
         else:
             self.logger.info(f"No failures for {job.name} #{job.build_id} were found!")
+            if success_rule := [rule for rule in firewatch_config.rules if rule.job_success is True]:  # type: ignore
+                self.logger.info(f"Reporting job {job.name} success in jira")
+                firewatch_config.jira.create_issue(
+                    project=success_rule[0].jira_project,  # type: ignore
```
Comment (myakove) [inline cli/report/report.py:66]: why `# type: ignore` is needed here?

File: cli/report/report.py [hunk 5]
```diff
@@ -60,6 +60,19 @@ def __init__(self, firewatch_config: Configuration, job: Job) -> None:
...
+                    summary=f"Success in {job.name}",
```
Comment (myakove) [inline cli/report/report.py:67]: maybe `f"{job.name} passed"` ?

File: cli/report/report.py [hunk 6]
```diff
@@ -532,3 +545,25 @@ def _get_open_bugs(
             return open_bugs
         else:
             return None
+
+    def _get_success_issue_description(
```
Comment (myakove) [inline cli/report/report.py:549]: Do we need another function for this?
`_get_issue_description` can get `success` arg and build the issue to match it

File: cli/report/report.py [hunk 7]
```diff
@@ -532,3 +545,25 @@ def _get_open_bugs(
...
+            build_id (Optional[str]): Build ID
+
+        Returns:
+            str: String object representing the description.
+        """
+        return f"""
+                    *Link:* https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/{job_name}/{build_id}
+                    *Build ID:* {build_id}
+
+                    This bug was filed using [firewatch in OpenShift CI|https://github.com/CSPI-QE/firewatch)]
```
Comment (myakove) [inline cli/report/report.py:568]: `This bug was` , This is success job, no bugs found

**Review by myakove — CHANGES_REQUESTED** [review]

File: cli/objects/jira_base.py (hunk 2 above)
Comment (rnetser) [reply cli/objects/jira_base.py:150]: appears in description

File: cli/report/__init__.py (hunk 3 above)
Comment (rnetser) [reply cli/report/__init__.py:102]: will remove and open an issue

**Review by myakove — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -39,18 +40,29 @@ Remember, when you are using the `firewatch-report-issues` ref, some variables n
...
-    FIREWATCH_CONFIG: |
-        [
-            {"step": "exact-step-name", "failure_type": "pod_failure", "classification": "Infrastructure", "jira_project": "PROJECT", "jira_component": "some-component", "jira_assignee": "some-user@redhat.com"},
-            {"step": "*partial-name*", "failure_type": "all", "classification":  "Misc.", "jira_project": "OTHER", "jira_component": ["component-1", "component-2"], "jira_priority": "major", "group": {"name": "some-group", "priority": 1}},
-            {"step": "*ends-with-this", "failure_type": "test_failure", "classification": "Test failures", "jira_project": "TEST", "jira_epic": "EPIC-123", "jira_additional_labels": ["test-label-1", "test-label-2"], "group": {"name": "some-group", "priority": 2}},
-            {"step": "*ignore*", "failure_type": "test_failure", "classification": "NONE", "jira_project": "NONE", "ignore": "true"},
-            {"step": "affects-version", "failure_type": "all", "classification": "Affects Version", "jira_project": "TEST", "jira_epic": "EPIC-123", "jira_affects_version": "4.14"}
-        ]
-    ```
+  - Examples:
```
Comment (calebevans) [inline README.md:43]:
```suggestion
  - Example:

    ```yaml
    FIREWATCH_CONFIG: |
        [
            {"step": "exact-step-name", "failure_type": "pod_failure", "classification": "Infrastructure", "jira_project": "PROJECT", "jira_component": "some-component", "jira_assignee": "some-user@redhat.com"},
            {"step": "*partial-name*", "failure_type": "all", "classification":  "Misc.", "jira_project": "OTHER", "jira_component": ["component-1", "component-2"], "jira_priority": "major", "group": {"name": "some-group", "priority": 1}},
            {"step": "*ends-with-this", "failure_type": "test_failure", "classification": "Test failures", "jira_project": "TEST", "jira_epic": "EPIC-123", "jira_additional_labels": ["test-label-1", "test-label-2"], "group": {"name": "some-group", "priority": 2}},
            {"step": "*ignore*", "failure_type": "test_failure", "classification": "NONE", "jira_project": "NONE", "ignore": "true"},
            {"step": "affects-version", "failure_type": "all", "classification": "Affects Version", "jira_project": "TEST", "jira_epic": "EPIC-123", "jira_affects_version": "4.14"},
            {"job_success": true, "jira_project": "TEST", "jira_epic": "EPIC-123"}
        ]
    ```
```

I would just add the option to the example and add a new section to the `docs/configuration_guide.md` for this

**Review by calebevans — COMMENTED** [review]
After the change to the docs, as long as this has been tested, it is okay with me. Thanks for the PR!

**Pull request merged**